src/core/utils/branches.tsx

```diff
@@ -35,9 +34,20 @@ const useGetCleanBranches = () => {
     branches,
     blacklistBranches
   );
-  const cleanBranches = cleanBranchesId(whitelistBranches);
+  return whitelistBranches;
+};
 
+const useGetCleanBranches = () => {
+  const config = useConfig();
+  const branches = getBranches(config);
+  const cleanBranches = cleanBranchesId(branches);
   return cleanBranches;
 };
 
+export const useGetBranches = () => {
+  const config = useConfig();
+  const branches = getBranches(config);
+  return branches;
+};
+
```

Comment:
As far as I can see you can merge getBranches() into this and use this function
and replace line 41 and 42 with calling: `useGetBranches()`.